### PR TITLE
emulationstation: refactor

### PR DESCRIPTION
**Pull request by AndersonTorres**
Found by https://github.com/NixOS/nixpkgs/commit/af039eb7c612e55cb7ed3141c3ae20587aafcf0f#r143796136

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/em/emulationstation/package.nix [hunk 1]
```diff
@@ -1,31 +1,32 @@
...
+  libGLU,
+  libvlc,
+  pkg-config,
+  rapidjson,
+  stdenv,
+  # Configurable options
+  sources ? callPackage ./sources.nix { },
 }:
 
 stdenv.mkDerivation (finalAttrs: {
```
Comment (kini) [inline pkgs/by-name/em/emulationstation/package.nix:21]: Perhaps `finalAttrs` is no longer needed?

Comment (AndersonTorres) [reply pkgs/by-name/em/emulationstation/package.nix:21]: Yes, but I see no necessity of removing it.

File: pkgs/by-name/em/emulationstation/sources.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+  fetchFromGitHub,
+}:
+
+{
+  emulationstation = let
```
Comment (JohnRTitor) [inline pkgs/by-name/em/emulationstation/sources.nix:6]: Why is there a need of this detachment?

Comment (AndersonTorres) [reply pkgs/by-name/em/emulationstation/sources.nix:6]: To not clutter the `package.nix` with 20+ lines of fetchers.

**Comment by AndersonTorres** [comment]
Since emulationstation needs an insecure freeimage (and I still did not figure out how to fix it), it will not be built by ofBorg.

File: pkgs/by-name/em/emulationstation/package.nix (hunk 1 above)
Comment (Aleksanaa) [reply pkgs/by-name/em/emulationstation/package.nix:21]: Unused attributes should be removed

File: pkgs/by-name/em/emulationstation/sources.nix (hunk 2 above)
Comment (JohnRTitor) [reply pkgs/by-name/em/emulationstation/sources.nix:6]: Fair enough, though there are way lengthy package definitions.

Might be a good idea to automate the updating.

Comment (AndersonTorres) [reply pkgs/by-name/em/emulationstation/sources.nix:6]: Yes, I know. Arcan uses four fetches afaicr.

File: pkgs/by-name/em/emulationstation/package.nix (hunk 1 above)
Comment (AndersonTorres) [reply pkgs/by-name/em/emulationstation/package.nix]: This is not an attribute, but an argument for `mkDerivation`.

File: pkgs/by-name/em/emulationstation/package.nix [hunk 3]
```diff
@@ -1,31 +1,32 @@
...
+  freeimage,
+  freetype,
+  libGL,
+  libGLU,
+  libvlc,
+  pkg-config,
+  rapidjson,
+  stdenv,
+  # Configurable options
+  sources ? callPackage ./sources.nix { },
```
[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/by-name/em/emulationstation/package.nix]: Done.

[the remark this answers is not in the document]
Comment (JohnRTitor) [reply pkgs/by-name/em/emulationstation/package.nix]: This accidentally got dropped?

[the remark this answers is not in the document]
Comment (AndersonTorres) [reply pkgs/by-name/em/emulationstation/package.nix]: No, it was on purpose.

**Review by superherointj — APPROVED** [review]
I find this external sources pattern unusual. Since you are maintainer of package. I won't dispute your choices on this. Code-wise it works, yes. Just weird.

**Comment by superherointj** [comment]
@edwtjo Do you have objections?

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**